server/src/main/java/org/opensearch/action/admin/cluster/remotestore/RemoteStoreNode.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.remotestore;
+
+import org.opensearch.cluster.metadata.RepositoriesMetadata;
+import org.opensearch.cluster.metadata.RepositoryMetadata;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.repositories.blobstore.BlobStoreRepository;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * This is an abstraction for validating and storing information specific to remote backed storage nodes.
+ *
+ * @opensearch.internal
+ */
+public class RemoteStoreNode {
+
+    public static final String REMOTE_STORE_NODE_ATTRIBUTE_KEY_PREFIX = "remote_store";
+    public static final String REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY = "remote_store.segment.repository";
+    public static final String REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY = "remote_store.translog.repository";
```

Comment:
These values needs to synced with [IndicesService](https://github.com/opensearch-project/OpenSearch/blob/8324b889b9da286f506451ff8133150e3a0b757b/server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectoryFactory.java#L46C1-L46C1)